app/docs/dev/policies/meshretry.md

```diff
@@ -0,0 +1,293 @@
+---
+title: MeshRetry (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [Retry](../retry).
+{% endwarning %}
+
+This policy enables {{site.mesh_product_name}} to know how to behave if there is a failed scenario (i.e. HTTP request) which could be retried.
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅        | ✅  | ❌   |
+| MeshSubset        | ✅        | ❌  | ❌   |
+| MeshService       | ✅        | ✅  | ❌   |
+| MeshServiceSubset | ✅        | ❌  | ❌   |
+| MeshGatewayRoute  | ❌        | ❌  | ❌   |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+The policy let you configure retry behaviour for `HTTP`, `GRPC` and `TCP` protocols.
+The protocol is selected by picking the most [specific protocol](/docs/{{ page.version }}/policies/protocol-support-in-kuma/#protocol-support-in-kuma).
+
+Each protocol has a separate section under `default` in the policy YAML.
+Some sections are common between protocols or have similar meaning.
+
+### Retry on
+
+The field `retryOn` is a list of conditions which will cause a retry.
+
+For `HTTP` these are related to the response status code or method ("5xx", "429", "HttpMethodGet").
+For `gRPC` these are status codes in response headers ("canceled", "deadline-exceeded", etc.).
+There is no equivalent for `TCP`.
+
+One or more conditions can be specified, for example:
+
+```yaml
+retryOn:
```

Comment:
Can we specify a full list of available values both for `HTTP` and `GRPC`? As we have in comments:
```go
// RetryOn is a list of conditions which will cause a retry. Available values are:
// [5XX, GatewayError, Reset, Retriable4xx, ConnectFailure, EnvoyRatelimited,
// RefusedStream, Http3PostConnectFailure, HttpMethodConnect, HttpMethodDelete,
// HttpMethodGet, HttpMethodHead, HttpMethodOptions, HttpMethodPatch,
// HttpMethodPost, HttpMethodPut, HttpMethodTrace].
// Also, any HTTP status code (500, 503, etc).
RetryOn *[]HTTPRetryOn `json:"retryOn,omitempty"`
```
and
```go
// RetryOn is a list of conditions which will cause a retry. Available values are:
// [Canceled, DeadlineExceeded, Internal, ResourceExhausted, Unavailable].
RetryOn *[]GRPCRetryOn `json:"retryOn,omitempty"`
```

Reply:
do you want it like this 👇 ?

Reply:
Yes, thanks!